desc/objectives/_profiles.py

```diff
@@ -264,6 +264,19 @@ def build(self, use_jit=True, verbose=1):
         else:
             grid = self._grid
 
+        warnif(
+            grid.num_theta < 2 * eq.M,
+            RuntimeWarning,
+            "RotationalTransform objective grid requires poloidal "
+            "resolution for surface averages",
+        )
+        warnif(
+            grid.num_zeta < 2 * eq.N,
+            RuntimeWarning,
+            "RotationalTransform objective grid requires toroidal "
```

Comment:
technically if an iota profile is assigned, then no surface averages are used right? Maybe add another conditional checking that eq.iota is None?